code-challenges.test.js

```diff
@@ -14,43 +14,120 @@
+        expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
 
 
 // b) Create the function that makes the test pass.
 
+// create a function name codedmessage that takes in a string
+// make an array theat stores that coded letters
+// make a conditional statement inside a loop where a string will be added if the word includes a letter of the coded message to the new variable of string
+// return the coded message
+
+function codedMessage(string) {
+    let arr = ["4", "3", "1", "0"]
+    let newString = ""
+    for (let i = 0; i < string.length; i++) {
+        if (string.charAt(i).toLowerCase() === "a") {
+            newString += arr[0]
+        } else if (string.charAt(i).toLowerCase() === "e") {
+            newString += arr[1]
+        } else if (string.charAt(i).toLowerCase() === "i") {
+            newString += arr[2]
+        } else if (string.charAt(i).toLowerCase() === "o") {
+            newString += arr[3]
+        } else {
+            newString += string[i]
+        }
+    }
+    return newString
+}
+
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
-
+describe("arrayContains", function(){
+    it("takes in an array of words and a single letter and returns all the words that contain that particular letter", function(){
+        const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    const letterA = "a"
+    // Expected output: ["Apple", "Banana", "Orange"]
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    const letterE = "e"
+    // Expected output: ["Cherry", "Blueberry", "Peach"]
+        expect(arrayContains(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"])
+        expect(arrayContains(arrayOfWords2, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
 
 // b) Create the function that makes the test pass.
 
+// make a function that have two parameters
+// use a filter method to filter out words that includes the character given
+// also use the lowercase method so that every letter will be equal to the given arguments
+// return the filtered words
+function arrayContains(arr, char) {
+    return arr.filter(a => {
+        return a.toLowerCase().includes(char)
+    })
+}
```

Comment:
Love it!  Would love to hear your thoughts on using function notation instead of arrow notation?